### DISCUSSION TITLE
Apply FWI for marmousi 2D model

### DISCUSSION THREAD

**Discussion by bening-gawitsa** [devitocodes/devito#2156] [Q&A]
I tried to apply FWI code example for marmousi 2D model using MPI with this configuration source and receiver

"nshots = 110  # Number of shots to create gradient from
nreceivers = 400  # Number of receiver locations per shot 
fwi_iterations = 5

# Define true and initial model
shape = (1601, 401)  # Number of grid point (nx, nz)
spacing = (7.5, 7.5)  # Grid spacing in m. The domain size is now 1km by 1km
origin = (0., 0.)  # Need origin to define relative source and receiver locations

#Acquisition geometry

#Define acquisition geometry: source
from examples.seismic import AcquisitionGeometry

t0 = 0.
tn = 7000. 
f0 = 0.015
# First, position source centrally in all dimensions, then set depth
src_coordinates = np.empty((nshots, 2), dtype=np.float32)
src_coordinates[:, 0] = 1.
src_coordinates[:, 0] = np.linspace(0, model.domain_size[0], num=nshots)

# Define acquisition geometry: receivers
# Initialize receivers for synthetic and imaging data

rec_coordinates = np.empty((nreceivers, 2))
rec_coordinates[:,1] = 200.
rec_coordinates[:,0] = np.linspace(0, model.domain_size[0], num=nreceivers) 

# Geometry

geometry = AcquisitionGeometry(model, rec_coordinates, src_coordinates, t0, tn, f0=f0, src_type='Ricker')
geometry.src.show()"

then I got this error message: "ValueError: Cannot increase dimensionality of MPI-distributed Data"

![Screenshot 2023-07-02 102517](https://github.com/devitocodes/devito/assets/63508383/0540dad2-f978-43ca-aa50-f2afb311b3a0)

could you help me how to solve it

**Comment by mloubout** [comment]
The error comes from `geometry.src.show()` because this is a fundamentally serial display utility. With `MPI` only one rank has a non-empty source so all the other rank will throw an error trying to plot the source wavelet. 
So I would recommend to debug without MPI with all the plots and such then you can run it with MPI commenting out these "serial" statements
